app/models/listing.rb

```diff
@@ -22,6 +22,10 @@ class Listing < ApplicationRecord
   after_commit :remove_from_elasticsearch, on: [:destroy]
   acts_as_taggable_on :tags
   has_many :credits, as: :purchase, inverse_of: :purchase, dependent: :nullify
+  has_many :listing_endorsements,
+            -> { where(approved: true) },
+            foreign_key: :classified_listing_id,
+            inverse_of: :listing
```

Comment:
we should add the `dependent: :destroy` clause so that if the listing is destroyed, the endorsements go with it. I would also call this just `endorsements`